src/wp-includes/media.php

```diff
@@ -240,7 +241,7 @@ function image_downsize( $id, $size = 'medium' ) {
 		$is_intermediate = true;
 	} elseif ( 'thumbnail' === $size && ! empty( $meta['thumb'] ) && is_string( $meta['thumb'] ) ) {
 		// Fall back to the old thumbnail.
-		$imagefile = get_attached_file( $id );
+		$imagefile = get_attached_file( $id, $mime_type );
```

Comment:
Related to my other comment on `get_attached_file()`, this is the only usage of `get_attached_file()` that uses the new parameter. I think we can simply remove this change from here though since the actual file name for `$imagefile` isn't even used in this function. So it doesn't really matter in which MIME type the file name is requested.